src/components/SlideElementWrapper.js

```diff
@@ -1,41 +1,74 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useSpring, animated } from 'react-spring';
-
-import { SlideContext } from '../hooks/useSlide';
+import { DeckContext } from '../hooks/useDeck';
+import { AnimationMutexContext } from '../hooks/useMutex';
 
 /**
  * SlideElementWrapper provides a component for animating slideElements
  * Anything wrapped inside will be affected by the transition.
  *
  * It is currently using useSpring but ideally we will be able to switch
  * to whatever react-spring hook a user desires!
- * 
+ *
  * Note: Immediate is a React-Spring property that we pass to the animations
  * essentially it skips animations.
  */
 
 const SlideElementWrapper = ({ elementNum, transitionEffect, children }) => {
-  const [state, ,] = React.useContext(SlideContext);
+  const [
+    { reverseDirection, currentSlideElement, immediateElement }
+  ] = React.useContext(DeckContext);
+  const { signal } = React.useContext(AnimationMutexContext);
+  const activeElement = elementNum === currentSlideElement;
+  const upcomingElement =
+    (elementNum > currentSlideElement && !reverseDirection) ||
+    (elementNum < currentSlideElement && reverseDirection);
+  const previousElement =
+    (elementNum < currentSlideElement && !reverseDirection) ||
+    (elementNum > currentSlideElement && reverseDirection);
```

Comment:
This component got a little more complex because we need to handle all the edge cases for slides when going forwards or backwards—or when direction changes midway.

React-spring does not commit changes in a permanent style to their nodes, so we have to know if they have already animated in or not (previousElement) so we can set the style props manually. Otherwise we would lose transitions on next element navigation. The conditions for upcoming and previous elements are reversed depending on the direction of the navigation.